### PR TITLE
python3Packages.datasets: 1.1.2 -> 1.4.1

### PR DESCRIPTION
**Pull request by danieldk**
###### Motivation for this change

Update datasets to 1.4.1.

https://github.com/huggingface/datasets/releases/tag/1.3.0
https://github.com/huggingface/datasets/releases/tag/1.4.0
https://github.com/huggingface/datasets/releases/tag/1.4.1

Since datasets now also requires Huggingface's hub package, this PR
also adds `python3Packages.huggingface_hub`.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 115237` at f8cb21b0 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/f022aff82dee96a5c334f09101de6156">python37Packages.datasets</a></li></li>
    <li><a href="https://gist.github.com/7cd325d5643ff3dc738fb9c6148631d5">python37Packages.huggingface_hub</a></li></li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/89e72c607dd33f4f4c63a49ab46b78e2">python38Packages.datasets</a></li></li>
    <li><a href="https://gist.github.com/1ba8b9fe8fd28ef45f8c72f7df0986e9">python38Packages.huggingface_hub</a></li></li>
  </ul>
</details>

**Comment by danieldk** [comment]
Fixed the Python 3.7 build.

File: pkgs/development/python-modules/datasets/default.nix [hunk 1]
```diff
@@ -3,6 +3,8 @@
 , fetchFromGitHub
 , dill
 , filelock
+, fsspec
+, huggingface_hub
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/datasets/default.nix:7]:
```suggestion
, huggingface-hub
```

File: pkgs/development/python-modules/huggingface_hub/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+, buildPythonPackage
+, pythonOlder
+, filelock
+, importlib-metadata
+, requests
+, tqdm
+}:
+
+buildPythonPackage rec {
+  pname = "huggingface_hub";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/huggingface_hub/default.nix:12]:
```suggestion
  pname = "huggingface-hub";
```

File: pkgs/development/python-modules/huggingface_hub/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  version = "0.0.6";
+
+  src = fetchFromGitHub {
+    owner = "huggingface";
+    repo = pname;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/huggingface_hub/default.nix:17]:
```suggestion
    repo = "huggingface_hub";
```

File: pkgs/development/python-modules/huggingface_hub/default.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  # Tests require network access.
+  doCheck = false;
+  pythonImportsCheck = [ "huggingface_hub" ];
+
+   meta = with lib; {
+    homepage = "https://github.com/huggingface/huggingface_hub";
+    description = "Download and publish models and other files on the huggingface.co hub";
+    changelog = "https://github.com/huggingface/huggingface_hub/releases/tag/${version}";
+    license = licenses.asl20;
+    platforms = platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/huggingface_hub/default.nix:37]:
```suggestion
```

File: pkgs/top-level/python-packages.nix [hunk 5]
```diff
@@ -3072,6 +3072,8 @@ in {
 
   hug = callPackage ../development/python-modules/hug { };
 
+  huggingface_hub = callPackage ../development/python-modules/huggingface_hub { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:3075]:
```suggestion
  huggingface-hub = callPackage ../development/python-modules/huggingface-hub { };
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/datasets/default.nix (hunk 1 above)
Comment (danieldk) [reply pkgs/development/python-modules/datasets/default.nix:7]: I considered that, but `huggingface_hub` is actually the upstream name.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 115237` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>python37Packages.datasets</li>
<li>python37Packages.huggingface_hub</li>
<li>python38Packages.datasets</li>
<li>python38Packages.huggingface_hub</li>
</ul>
</details>

File: pkgs/development/python-modules/datasets/default.nix (hunk 1 above)
Comment (danieldk) [reply pkgs/development/python-modules/datasets/default.nix:7]: Not sure what the policy is in such cases.

Comment (SuperSandro2000) [reply pkgs/development/python-modules/datasets/default.nix:7]: We normalize the package names according to some PEP which replaces special characters with like _ with -.

Comment (danieldk) [reply pkgs/development/python-modules/datasets/default.nix]: Ack, all fixed.
